test/tls.js

```diff
@@ -116,6 +116,26 @@ describe('TLS', () => {
     })
   })
 
+  it('should not timeout when it tls connects', done => {
```

Comment:
Is that a test for the original issue? You set timeout of 1sec with TLS connection, but how do we check here that it would be failing (if the bug was there)? With the bug what would be the state of the connection, connected, disconnected, none of the above? Should we create a sub and from another connection (without timeout) send a message after the timeout that would have cause the disconnect, and make sure it is received?

Reply:
just added a flush, which should accomplish the same.